### PR TITLE
Don't play blood effects for completely resisted hits

### PR DESCRIPTION
**Pull request by Allofich**
For https://bugs.openmw.org/issues/3549.

The check for playing the blood effect was being done before the weapon resist effect was applied, so the blood effect was showing when it shouldn't. Test with any creature or npc by giving them the "Immune to Normal Weapons" spell and hitting them with a weapon that doesn't get through this effect.

### CODE REVIEW

**Comment by Allofich** [comment]
Actually, I think it might be better to move the playing of the blood effect into onHit. I'm going to rewrite this.

**Comment by Allofich** [comment]
New version up.

Old version is at https://github.com/Allofich/openmw/tree/onhit2. It made onHit() return damage (after resistance was applied to it) as a float, which was then used to determine whether to play the blood effect, but I think the new version is better.

File: apps/openmw/mwclass/creature.cpp [hunk 1]
```diff
@@ -318,15 +318,12 @@ namespace MWClass
...
-            MWBase::Environment::get().getWorld()->spawnBloodEffect(victim, hitPosition);
-
         MWMechanics::diseaseContact(victim, ptr);
 
-        victim.getClass().onHit(victim, damage, healthdmg, weapon, ptr, true);
+        victim.getClass().onHit(victim, damage, healthdmg, weapon, ptr, hitPosition, true);
     }
 
-    void Creature::onHit(const MWWorld::Ptr &ptr, float damage, bool ishealth, const MWWorld::Ptr &object, const MWWorld::Ptr &attacker, bool successful) const
+    void Creature::onHit(const MWWorld::Ptr &ptr, float damage, bool ishealth, const MWWorld::Ptr &object, const MWWorld::Ptr &attacker, osg::Vec3f hitPosition, bool successful) const
```
Comment (scrawl) [inline apps/openmw/mwclass/creature.cpp]: hitPosition could be passed by const reference: ```, const osg::Vec3f& hitPosition, ```

**Comment by Allofich** [comment]
Let me also make them pass as reference.

**Pull request merged**